From patchwork Thu Oct  4 02:04:22 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Wei Yang <richard.weiyang@gmail.com>
X-Patchwork-Id: 10625473
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 6B3C8112B
	for <patchwork-kvm@patchwork.kernel.org>;
 Thu,  4 Oct 2018 02:04:50 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 59FB529075
	for <patchwork-kvm@patchwork.kernel.org>;
 Thu,  4 Oct 2018 02:04:50 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 4DBC42907B; Thu,  4 Oct 2018 02:04:50 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-8.0 required=2.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,DKIM_VALID_AU,FREEMAIL_FROM,MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI
	autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id EB7A429075
	for <patchwork-kvm@patchwork.kernel.org>;
 Thu,  4 Oct 2018 02:04:49 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1726843AbeJDIzs (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Thu, 4 Oct 2018 04:55:48 -0400
Received: from mail-pf1-f194.google.com ([209.85.210.194]:36028 "EHLO
        mail-pf1-f194.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1726735AbeJDIzr (ORCPT <rfc822;kvm@vger.kernel.org>);
        Thu, 4 Oct 2018 04:55:47 -0400
Received: by mail-pf1-f194.google.com with SMTP id b7-v6so2574227pfo.3
        for <kvm@vger.kernel.org>; Wed, 03 Oct 2018 19:04:48 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=gmail.com; s=20161025;
        h=from:to:cc:subject:date:message-id:in-reply-to:references;
        bh=+ksXHBEwYT1zs0C4zpAYb/1E2EJoQo/a0dWNenDqb/g=;
        b=IEKefqzSzn5H8TIK2kgUnL98dqPHJm24JEVVxuliMSbipVVMaPHBsdh4Yqsz+Z8MKt
         z49EutgWgHnqwEbPNvRMRz91UX8e8h5mdxgbDTNGUOxdIWamNPclaMX07kNkFwVtf5Jb
         jLL5RBZK8Gyh+Bsa9Ui9F7zFuR7Wp6jgKN3X16JW+a6no6obWVYIaQe5GJN9YsODr+QP
         TO0ZUhUvTC/RRG9iOl6DbdMNsh0MXnsiuimcsK7nzvS9FcdPuGke2zUVRk+o4+it6atU
         2UYS/sfp5RrEyc08nwp2qfVDol+ODmR0xXAcj32z1HkEfEDASqo4HCo/G4D5rRImiHIx
         rmBQ==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
         :references;
        bh=+ksXHBEwYT1zs0C4zpAYb/1E2EJoQo/a0dWNenDqb/g=;
        b=Pu52xrtWbZE4JAM9RF3eQaic987mkvV3men4nY/iUGbpdtrxuDJ/+1GGg0ke9C2UJf
         RHMclpC55bgL/cJKrAXjCmNE725soixZJNRWleeAU2H1dvhEy8hWBUvUufrCgjXa+FTv
         Jeyk5xvB9+nTNuJP0y5go+utj7lLswLDZ1Yud1y2FaDy4DjAtmWMAB8bIUKQUnsYAJD5
         HQqH6PImp+lEWxQp5mxxrWYjX+vaQsXfkYILGWSntIOsNXLMq/VnFunxGzRFtCW84HHl
         +EUcZzf8aumiOIeWjUnC6B56GZ8pMX/44ywlWLyCZrOS1uSrpIBcYeJhAcXUsjeQyLJe
         Ipkw==
X-Gm-Message-State: ABuFfohfOtw1ug1loeXFQ7bJybPqb/ybEDyAY0WkfA6hfAv13wbNgM6h
        oRxInpNQPsy07cF4lFnJnoo=
X-Google-Smtp-Source: 
 ACcGV63vK50t9AD71mG7slNAb0bRqsOBDGMoEjQlBCzgJzFcf1UiXvoRvnYkulb+lBK/RRzTF7bB+w==
X-Received: by 2002:a62:444f:: with SMTP id
 r76-v6mr4480754pfa.111.1538618687922;
        Wed, 03 Oct 2018 19:04:47 -0700 (PDT)
Received: from localhost ([185.92.221.13])
        by smtp.gmail.com with ESMTPSA id
 z15-v6sm3947874pfn.170.2018.10.03.19.04.46
        (version=TLS1_2 cipher=ECDHE-RSA-AES128-GCM-SHA256 bits=128/128);
        Wed, 03 Oct 2018 19:04:47 -0700 (PDT)
From: Wei Yang <richard.weiyang@gmail.com>
To: pbonzini@redhat.com, rkrcmar@redhat.com
Cc: x86@kernel.org, kvm@vger.kernel.org,
        Wei Yang <richard.weiyang@gmail.com>
Subject: [Patch v3 1/3] KVM: x86: rename pte_list_remove to __pte_list_remove
Date: Thu,  4 Oct 2018 10:04:22 +0800
Message-Id: <20181004020424.96171-2-richard.weiyang@gmail.com>
X-Mailer: git-send-email 2.15.1
In-Reply-To: <20181004020424.96171-1-richard.weiyang@gmail.com>
References: <20180926075853.28715-1-richard.weiyang@gmail.com>
 <20181004020424.96171-1-richard.weiyang@gmail.com>
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

This is a patch preparing for further change.

Signed-off-by: Wei Yang <richard.weiyang@gmail.com>
---
 arch/x86/kvm/mmu.c | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/arch/x86/kvm/mmu.c b/arch/x86/kvm/mmu.c
index 9dca99948e12..de673c073cf0 100644
--- a/arch/x86/kvm/mmu.c
+++ b/arch/x86/kvm/mmu.c
@@ -1249,24 +1249,24 @@ pte_list_desc_remove_entry(struct kvm_rmap_head *rmap_head,
 	mmu_free_pte_list_desc(desc);
 }
 
-static void pte_list_remove(u64 *spte, struct kvm_rmap_head *rmap_head)
+static void __pte_list_remove(u64 *spte, struct kvm_rmap_head *rmap_head)
 {
 	struct pte_list_desc *desc;
 	struct pte_list_desc *prev_desc;
 	int i;
 
 	if (!rmap_head->val) {
-		printk(KERN_ERR "pte_list_remove: %p 0->BUG\n", spte);
+		pr_err("%s: %p 0->BUG\n", __func__, spte);
 		BUG();
 	} else if (!(rmap_head->val & 1)) {
-		rmap_printk("pte_list_remove:  %p 1->0\n", spte);
+		rmap_printk("%s:  %p 1->0\n", __func__, spte);
 		if ((u64 *)rmap_head->val != spte) {
-			printk(KERN_ERR "pte_list_remove:  %p 1->BUG\n", spte);
+			pr_err("%s:  %p 1->BUG\n", __func__, spte);
 			BUG();
 		}
 		rmap_head->val = 0;
 	} else {
-		rmap_printk("pte_list_remove:  %p many->many\n", spte);
+		rmap_printk("%s:  %p many->many\n", __func__, spte);
 		desc = (struct pte_list_desc *)(rmap_head->val & ~1ul);
 		prev_desc = NULL;
 		while (desc) {
@@ -1280,7 +1280,7 @@ static void pte_list_remove(u64 *spte, struct kvm_rmap_head *rmap_head)
 			prev_desc = desc;
 			desc = desc->more;
 		}
-		pr_err("pte_list_remove: %p many->many\n", spte);
+		pr_err("%s: %p many->many\n", __func__, spte);
 		BUG();
 	}
 }
@@ -1333,7 +1333,7 @@ static void rmap_remove(struct kvm *kvm, u64 *spte)
 	sp = page_header(__pa(spte));
 	gfn = kvm_mmu_page_get_gfn(sp, spte - sp->spt);
 	rmap_head = gfn_to_rmap(kvm, gfn, sp);
-	pte_list_remove(spte, rmap_head);
+	__pte_list_remove(spte, rmap_head);
 }
 
 /*
@@ -1977,7 +1977,7 @@ static void mmu_page_add_parent_pte(struct kvm_vcpu *vcpu,
 static void mmu_page_remove_parent_pte(struct kvm_mmu_page *sp,
 				       u64 *parent_pte)
 {
-	pte_list_remove(parent_pte, &sp->parent_ptes);
+	__pte_list_remove(parent_pte, &sp->parent_ptes);
 }
 
 static void drop_parent_pte(struct kvm_mmu_page *sp,

From patchwork Thu Oct  4 02:04:23 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Wei Yang <richard.weiyang@gmail.com>
X-Patchwork-Id: 10625475
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 3CB15112B
	for <patchwork-kvm@patchwork.kernel.org>;
 Thu,  4 Oct 2018 02:04:59 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 2982029080
	for <patchwork-kvm@patchwork.kernel.org>;
 Thu,  4 Oct 2018 02:04:59 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 1DEBB29091; Thu,  4 Oct 2018 02:04:59 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-8.0 required=2.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,DKIM_VALID_AU,FREEMAIL_FROM,MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI
	autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id B34BD29087
	for <patchwork-kvm@patchwork.kernel.org>;
 Thu,  4 Oct 2018 02:04:58 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1727264AbeJDIz4 (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Thu, 4 Oct 2018 04:55:56 -0400
Received: from mail-pl1-f196.google.com ([209.85.214.196]:42291 "EHLO
        mail-pl1-f196.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1726949AbeJDIz4 (ORCPT <rfc822;kvm@vger.kernel.org>);
        Thu, 4 Oct 2018 04:55:56 -0400
Received: by mail-pl1-f196.google.com with SMTP id c8-v6so4353444plo.9
        for <kvm@vger.kernel.org>; Wed, 03 Oct 2018 19:04:57 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=gmail.com; s=20161025;
        h=from:to:cc:subject:date:message-id:in-reply-to:references;
        bh=sVghiJuB8JDPJYEUWgxC7oJ04V54N9+qABIcm8xF990=;
        b=Pj/KdFAWFynTPJxAouTb1iAl1BbNqcy/WuMYZ7PvEiiPdM6f5srU5gzrFyQHkuWVnc
         a/Lyu4ZzkOJVEL/nuJakwUD9MCcYE2F6WyS6DEG8O9KP2HxMN86meUAHkC5SJPBR/pYB
         dWnk9wRFt1+HJcRB4D79Ex1c/AmW5qZnziob5g/gVLGShYRFhJBpKqGAU7Eqd1y6YUus
         eTBcJfVX8WoBr4j9v1MO611T/e+uYgOwz4FGyp8luKrss02JEt37hASyPGd8LXZKn/bL
         6KRljdBmojxCpzE3ttxGXD/T0amQuXOb0CPiL6OBSufjmkCA9uZdeiu8yZXzYI26QHZQ
         13iA==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
         :references;
        bh=sVghiJuB8JDPJYEUWgxC7oJ04V54N9+qABIcm8xF990=;
        b=I/PZMK9mO85lW0hIEM/tOeg58PkGHOfsDvptMGmNyTPP3NRFGdrHsMLFQKlVcvcAlc
         AH1E0txmMif/m4LyU33AYcVoX93R5+lV7EZztQljNA/aiWedA9RYJtZdkKOPEPggtyrI
         WiXTPuQSDS0nINZE4xwUQ6/DobXKKF9rpqq+uj7fw9NaJBGt2ENRFyT6AMA4ryb6cdZ8
         av2hchA2fpJmCGputapBo419NeOUhtzMj4EhU5/l3vrxuNNvvmLbfz49moWsUsMUvbHH
         sMErw0y/QSQlrLd2qrgBGWHajzuBojo6/s4rwJaMskGd4W3sQfAhC3B8nViL+XoJ8LMy
         5f5g==
X-Gm-Message-State: ABuFfoi3JBR+zbmqrMKJN5kWPyP+/vyKFncwoE6NckhEh2q7q/QbQqOR
        nyFRxCjDacHCn1uyfqyXydI=
X-Google-Smtp-Source: 
 ACcGV61OZGkSRyFu1bbfM+oe6IRmtyRCHcxsy3iZsVE2KJxcZDmFC5XKsSmz1D2mrXYD5PB6zARTFA==
X-Received: by 2002:a17:902:8eca:: with SMTP id
 x10-v6mr4305944plo.336.1538618697048;
        Wed, 03 Oct 2018 19:04:57 -0700 (PDT)
Received: from localhost ([185.92.221.13])
        by smtp.gmail.com with ESMTPSA id
 z15-v6sm3948147pfn.170.2018.10.03.19.04.55
        (version=TLS1_2 cipher=ECDHE-RSA-AES128-GCM-SHA256 bits=128/128);
        Wed, 03 Oct 2018 19:04:56 -0700 (PDT)
From: Wei Yang <richard.weiyang@gmail.com>
To: pbonzini@redhat.com, rkrcmar@redhat.com
Cc: x86@kernel.org, kvm@vger.kernel.org,
        Wei Yang <richard.weiyang@gmail.com>
Subject: [Patch v3 2/3] KVM: x86: a new pte_list_remove including
 mmu_spte_clear_track_bits
Date: Thu,  4 Oct 2018 10:04:23 +0800
Message-Id: <20181004020424.96171-3-richard.weiyang@gmail.com>
X-Mailer: git-send-email 2.15.1
In-Reply-To: <20181004020424.96171-1-richard.weiyang@gmail.com>
References: <20180926075853.28715-1-richard.weiyang@gmail.com>
 <20181004020424.96171-1-richard.weiyang@gmail.com>
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

rmap_remove() would remove the sptep after locating the correct rmap_head.
While in several cases, the caller has already known the correct rmap_head.

This patch introduces a new pte_list_remove() which removes sptep directly
from rmap_head after clearing the track bits. Since it is for sure the spte
is present, it is safe to do the removal.

Signed-off-by: Wei Yang <richard.weiyang@gmail.com>
---
 arch/x86/kvm/mmu.c | 12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

diff --git a/arch/x86/kvm/mmu.c b/arch/x86/kvm/mmu.c
index de673c073cf0..5b05a8079872 100644
--- a/arch/x86/kvm/mmu.c
+++ b/arch/x86/kvm/mmu.c
@@ -1285,6 +1285,12 @@ static void __pte_list_remove(u64 *spte, struct kvm_rmap_head *rmap_head)
 	}
 }
 
+static void pte_list_remove(struct kvm_rmap_head *rmap_head, u64 *sptep)
+{
+	mmu_spte_clear_track_bits(sptep);
+	__pte_list_remove(sptep, rmap_head);
+}
+
 static struct kvm_rmap_head *__gfn_to_rmap(gfn_t gfn, int level,
 					   struct kvm_memory_slot *slot)
 {
@@ -1669,7 +1675,7 @@ static bool kvm_zap_rmapp(struct kvm *kvm, struct kvm_rmap_head *rmap_head)
 	while ((sptep = rmap_get_first(rmap_head, &iter))) {
 		rmap_printk("%s: spte %p %llx.\n", __func__, sptep, *sptep);
 
-		drop_spte(kvm, sptep);
+		pte_list_remove(rmap_head, sptep);
 		flush = true;
 	}
 
@@ -1705,7 +1711,7 @@ static int kvm_set_pte_rmapp(struct kvm *kvm, struct kvm_rmap_head *rmap_head,
 		need_flush = 1;
 
 		if (pte_write(*ptep)) {
-			drop_spte(kvm, sptep);
+			pte_list_remove(rmap_head, sptep);
 			goto restart;
 		} else {
 			new_spte = *sptep & ~PT64_BASE_ADDR_MASK;
@@ -5598,7 +5604,7 @@ static bool kvm_mmu_zap_collapsible_spte(struct kvm *kvm,
 		if (sp->role.direct &&
 			!kvm_is_reserved_pfn(pfn) &&
 			PageTransCompoundMap(pfn_to_page(pfn))) {
-			drop_spte(kvm, sptep);
+			pte_list_remove(rmap_head, sptep);
 			need_tlb_flush = 1;
 			goto restart;
 		}

From patchwork Thu Oct  4 02:04:24 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Wei Yang <richard.weiyang@gmail.com>
X-Patchwork-Id: 10625477
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 80D51112B
	for <patchwork-kvm@patchwork.kernel.org>;
 Thu,  4 Oct 2018 02:05:08 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 6F6FF29080
	for <patchwork-kvm@patchwork.kernel.org>;
 Thu,  4 Oct 2018 02:05:08 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 637122908A; Thu,  4 Oct 2018 02:05:08 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-8.0 required=2.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,DKIM_VALID_AU,FREEMAIL_FROM,MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI
	autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 1A22C29080
	for <patchwork-kvm@patchwork.kernel.org>;
 Thu,  4 Oct 2018 02:05:08 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1727283AbeJDI4G (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Thu, 4 Oct 2018 04:56:06 -0400
Received: from mail-pf1-f193.google.com ([209.85.210.193]:36976 "EHLO
        mail-pf1-f193.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1726735AbeJDI4G (ORCPT <rfc822;kvm@vger.kernel.org>);
        Thu, 4 Oct 2018 04:56:06 -0400
Received: by mail-pf1-f193.google.com with SMTP id j23-v6so2571964pfi.4
        for <kvm@vger.kernel.org>; Wed, 03 Oct 2018 19:05:06 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=gmail.com; s=20161025;
        h=from:to:cc:subject:date:message-id:in-reply-to:references;
        bh=QnnMnbd54dTBdmqIZJ48pt/0quLif9Agk/0crEiTNGE=;
        b=YVvaDmI9X2pLeyNowlYmAREImlCNLi0YUP8U5NVi8v2vca7ZJqNgBAUuT7SirRow+e
         Ejv72jAP8K5aSA5JkmMDdO8oU2f8eWZknpjp1X5k6DHGL9ETM6VkWDj2ONJaB/6yLrTk
         vwUGfoZjgUlQUIJk5JYQYSDTicNEFvCP2VttUto6OKfyeYs39Nta5/yM9HWjopMhXg3u
         5CEdpH6f+kbAHCN2sjXtu5KaGHHdkw5KtTlUwBzni4oz9wMiUhYwY1Y5BH6MayOUwago
         UjsmbfeFwOxEj83aFY5AYdswqVkZK3Q1WPIat4pRJOTYT+nmBIxuJKHYF0sU+G0Bo6jN
         DIkQ==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
         :references;
        bh=QnnMnbd54dTBdmqIZJ48pt/0quLif9Agk/0crEiTNGE=;
        b=m3pea7O+p13U/rVrUyxBTgTLji7LvmyMcPz5Nj/+PQSib1ZaEyinZ9Nr+F5ScOubLP
         AfbCnvvbqj+YT/2KFl5ftxvY19pArkwoDR92rqjPIwrKeMYU4YsdQd/o9wSdLlmpID46
         P0yN3Ls5SMPqRVkG2wjPP0wJGq/ooCk9mjkacvnbU3jW+nVXee9TewaAoYrAMexDqEKc
         rvsI4zQqNZB0h4o0qgxMS/X7XCEpgJg+mFNzA6lf/b3yLsDfqYM93mXAhwN9smzI9t/Z
         1tEOwWgcogiuxMKp1qXc9GdauGLShkQ3KQzv9WQBN1nqWleGGt4fjzfhvHkaMCiDPuVQ
         YNrA==
X-Gm-Message-State: ABuFfognPBh1Um3TBkN899nRi7jeyj4fDUoppKvVdA8hIILjF6K6S9Gs
        QBBzDmLWplXZYhQPMk/L/aEwIsSV
X-Google-Smtp-Source: 
 ACcGV62nAQORsG/u79UnyZxQFl4vYQLZ8VWiS5/qfD9SDbE0bPh3FELsRNKUq5Bs6WpNCwtXli18Tw==
X-Received: by 2002:a63:d34f:: with SMTP id
 u15-v6mr3720022pgi.325.1538618706420;
        Wed, 03 Oct 2018 19:05:06 -0700 (PDT)
Received: from localhost ([185.92.221.13])
        by smtp.gmail.com with ESMTPSA id
 e3-v6sm3787641pgc.71.2018.10.03.19.05.05
        (version=TLS1_2 cipher=ECDHE-RSA-AES128-GCM-SHA256 bits=128/128);
        Wed, 03 Oct 2018 19:05:05 -0700 (PDT)
From: Wei Yang <richard.weiyang@gmail.com>
To: pbonzini@redhat.com, rkrcmar@redhat.com
Cc: x86@kernel.org, kvm@vger.kernel.org,
        Wei Yang <richard.weiyang@gmail.com>
Subject: [Patch v3 3/3] KVM: x86: extract mmu_spte_clear_track_bits in the
 loop of kvm_set_pte_rmapp
Date: Thu,  4 Oct 2018 10:04:24 +0800
Message-Id: <20181004020424.96171-4-richard.weiyang@gmail.com>
X-Mailer: git-send-email 2.15.1
In-Reply-To: <20181004020424.96171-1-richard.weiyang@gmail.com>
References: <20180926075853.28715-1-richard.weiyang@gmail.com>
 <20181004020424.96171-1-richard.weiyang@gmail.com>
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

In the loop of kvm_set_pte_rmapp(), both branch of if clause would call
mmu_spte_clear_track_bits().

  * one indirectly by pte_list_remove in *if* branch
  * one directly in *else* branch

This patch extracts this behavior and call it before if clause.

Signed-off-by: Wei Yang <richard.weiyang@gmail.com>
---
 arch/x86/kvm/mmu.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/x86/kvm/mmu.c b/arch/x86/kvm/mmu.c
index 5b05a8079872..edf3d51b485c 100644
--- a/arch/x86/kvm/mmu.c
+++ b/arch/x86/kvm/mmu.c
@@ -1709,9 +1709,10 @@ static int kvm_set_pte_rmapp(struct kvm *kvm, struct kvm_rmap_head *rmap_head,
 			    sptep, *sptep, gfn, level);
 
 		need_flush = 1;
+		mmu_spte_clear_track_bits(sptep);
 
 		if (pte_write(*ptep)) {
-			pte_list_remove(rmap_head, sptep);
+			__pte_list_remove(sptep, rmap_head);
 			goto restart;
 		} else {
 			new_spte = *sptep & ~PT64_BASE_ADDR_MASK;
@@ -1722,7 +1723,6 @@ static int kvm_set_pte_rmapp(struct kvm *kvm, struct kvm_rmap_head *rmap_head,
 
 			new_spte = mark_spte_for_access_track(new_spte);
 
-			mmu_spte_clear_track_bits(sptep);
 			mmu_spte_set(sptep, new_spte);
 		}
 	}
